lib/ddtrace/contrib/action_mailer/events/process.rb

```diff
@@ -0,0 +1,39 @@
+require 'ddtrace/contrib/action_mailer/ext'
+require 'ddtrace/contrib/action_mailer/event'
+
+module Datadog
+  module Contrib
+    module ActionMailer
+      module Events
+        # Defines instrumentation for process.action_mailer event
+        module Process
+          include ActionMailer::Event
+
+          EVENT_NAME = 'process.action_mailer'.freeze
+
+          module_function
+
+          def event_name
+            self::EVENT_NAME
+          end
+
+          def span_name
+            Ext::SPAN_PROCESS
+          end
+
+          def span_type
+            # ActionMailer creates emails like a controller
```

Comment:
Could you clarify this? Are these top-level spans then? Do they happen on the same execution context as, say, a Rack request?

Reply:
i believe they are _not_ top level spans (although, if executed from say, the rails console i suppose they could be) but i am going to try to stress test this in a sample app, will report back when i have more. tl;dr i believe i've botched the span_type here and need to update, but want to confirm that

Reply:
So, to update: So this integration is similar to active record or action view in that there's a few different spans which could all be "top level" in the sense that they're the `1st` span for the service, but generally speaking they'll rarely be the _root_ span of the trace, usualy they're a child of action_pack controller span or a job consumer (delayed job, sidekiq, etc) span.

Here's a few examples:

1. async job queue (deliver_later)

<img width="893" alt="Image 2021-07-29 at 4 01 30 PM" src="https://user-images.githubusercontent.com/14250318/128391517-6e559df0-b40a-46dd-a198-4c15ae7d410b.png">

2. syncronous, no job queue (deliver_now)

<img width="900" alt="Image 2021-07-29 at 1 51 29 PM" src="https://user-images.githubusercontent.com/14250318/128391773-c3a80f14-2a24-4cd9-8f2e-0ea617d058bf.png">

I've tried to update the span_types to reflect existing standards that are closer to their more appropriate usage, marking them as either a `template` (which is what `.process` spans do, they render the mailer templates) or `worker` (which is what `.deliver` spans do, they send emails).

Anyway lmk what you think

Reply:
Makes sense. I like the traces you showed above, they all seem reasonable to me 👍.